runtime/vm/jvminit.c

```diff
@@ -719,6 +719,12 @@ freeJavaVM(J9JavaVM * vm)
 		vm->jniGlobalReferences = NULL;
 	}
 
+#if !defined(J9ZOS390)
```

Comment:
Why the ifdef? If anything, I'd expect to only do this on ZOS, not everywhere but. The fields aren't ifdeffed.

Reply:
this is due to the comment starting at line 746.

```
                /* zOS: Do not close any of the DLLs. This is necessary 
                 * because we do not know for sure whether all the threads 
                 * that may depend on these DLLs have terminated.
                 *
                 * Note that this solution is also suggested in the 
                 * 'z/OS XL C/C++ Programming Guide' V1R13 (Chapter 21 
                 * under the heading 'DLL Restrictions').
                 */
#if !defined(J9ZOS390)
                closeAllDLLs(vm);
                /* Remember the file descriptor of the trace DLL. This has to be closed later than other DLLs. */
                traceLoadInfo = FIND_DLL_TABLE_ENTRY(J9_RAS_TRACE_DLL_NAME);
                if (NULL != traceLoadInfo) {
                        traceDescriptor = traceLoadInfo->descriptor;
                }    
#endif /* !defined(J9ZOS390) */
```

Reply:
Good catch - please move the close code into `closeAllDLLs`